src/Avalanche/Signer.cpp

```diff
@@ -0,0 +1,353 @@
+// Copyright © 2017-2021 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+#include "Address.h"
+#include "../PublicKey.h"
+
+using namespace TW;
+using namespace TW::Avalanche;
+
+std::vector<Address> structToAddresses(const google::protobuf::internal::RepeatedStringTypeTraits::RepeatedFieldType inAddrs) {
+    std::vector<Address> out;
+    for (auto &addrBytes : inAddrs) {
+        auto pubkeyData = Data(addrBytes.begin(), addrBytes.end());
+        auto pubkey = PublicKey(pubkeyData, TWPublicKeyTypeSECP256k1);
+        auto addr = Address(pubkey);
+        out.push_back(addr);
+    }
+    return out;
+}
+
+std::vector<TransferableInput> structToInputs(const google::protobuf::RepeatedPtrField<TW::Avalanche::Proto::TransferableInput> inputStructs) noexcept {
+    std::vector<TransferableInput> inputs;
+    for (auto &inputStruct : inputStructs) {
+        auto txid = Data(inputStruct.tx_id().begin(), inputStruct.tx_id().end());
+        auto utxoIndex = inputStruct.utxo_index();
+        auto assetID = Data(inputStruct.asset_id().begin(), inputStruct.asset_id().end());
+        std::vector<Address> spendableAddresses = structToAddresses(inputStruct.spendable_addresses());
+
+        switch (inputStruct.input().input_case()) {
```

Comment:
warning: enumeration value 'INPUT_NOT_SET' not handled in switch
Switch with one branch, no default